From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Price <greg@zulip.com>
Date: Tue, 7 Jun 2022 21:08:51 -0700
Subject: [tsflower] Fix `import typeof` to `import type` for Flow-only types

I don't understand why TsFlower emitted `typeof` here; will have to
debug.

Probably a key part of the story is that although upstream imports
these as types, e.g.:

    import type { DirectEventHandler, Double } from 'react-native/Libraries/Types/CodegenTypes';

they don't actually exist in the TS type definitions in
`@types/react-native`.  They only exist as Flow definitions, in
`.js` files in RN itself.  This TsFlower behavior is what I'd expect
to see if when it asks the TS type-checker about these symbols, it
says that they *are* defined as values, but aren't as types.

(If that is the case, though, then I'd sure think these types aren't
doing any good in the upstream type definitions.  Perhaps they aren't?)
---
 .../lib/typescript/specs/NativeSafeAreaContext.js.flow       | 2 +-
 .../lib/typescript/specs/NativeSafeAreaProvider.js.flow      | 5 +----
 .../lib/typescript/specs/NativeSafeAreaView.js.flow          | 2 +-
 3 files changed, 3 insertions(+), 6 deletions(-)

diff --git types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaContext.js.flow types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaContext.js.flow
index 4972a87fc..80f3194b4 100644
--- types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaContext.js.flow
+++ types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaContext.js.flow
@@ -1,7 +1,7 @@
 /* @flow
  * @generated by TsFlower
  */
-import { typeof Double } from 'react-native/Libraries/Types/CodegenTypes';
+import { type Double } from 'react-native/Libraries/Types/CodegenTypes';
 import { typeof TurboModule } from 'react-native/Libraries/TurboModule/RCTExport';
 
 export interface Spec extends TurboModule {
diff --git types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaProvider.js.flow types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaProvider.js.flow
index 5d9e21df4..c7d784bd2 100644
--- types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaProvider.js.flow
+++ types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaProvider.js.flow
@@ -3,10 +3,7 @@
  */
 import type { Readonly } from 'tsflower/subst/lib';
 import type { ViewProps as $tsflower_subst$RN$ViewProps } from 'tsflower/subst/react-native';
-import {
-  typeof DirectEventHandler,
-  typeof Double,
-} from 'react-native/Libraries/Types/CodegenTypes';
+import { type DirectEventHandler, type Double } from 'react-native/Libraries/Types/CodegenTypes';
 import { type HostComponent } from 'react-native';
 
 export type Event = Readonly<{
diff --git types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaView.js.flow types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaView.js.flow
index 300c246bb..8095e858d 100644
--- types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaView.js.flow
+++ types/react-native-safe-area-context/lib/typescript/specs/NativeSafeAreaView.js.flow
@@ -2,7 +2,7 @@
  * @generated by TsFlower
  */
 import type { ViewProps as $tsflower_subst$RN$ViewProps } from 'tsflower/subst/react-native';
-import { typeof WithDefault } from 'react-native/Libraries/Types/CodegenTypes';
+import { type WithDefault } from 'react-native/Libraries/Types/CodegenTypes';
 import { type HostComponent } from 'react-native';
 
 export type NativeProps = {
-- 
2.32.0

